File: IsPalindrome.java
public class IsPalindrome {
    public static boolean isPalindrome(String s) {
        if (s.length() <= 1) return true;
        return s.charAt(0) == s.charAt(s.length() - 1) 
            && isPalindrome(s.substring(1, s.length()-1));
    }
    
    public static void main(String[] args) {
        System.out.println(isPalindrome("abcba"));
        System.out.println(isPalindrome("abccba"));
        System.out.println(isPalindrome("abcdba"));
        System.out.println(isPalindrome("abccbb"));
        System.out.println(isPalindrome("a"));
        System.out.println(isPalindrome(""));
    }
}

File: python3/IsPalindrome.py
class IsPalindrome:
	def is_palindrome(self, s):
		"""
		:type s: string
		:rtype: boolean
		"""
		if len(s) <= 1:
			return True
		return s[0] == s[-1] and self.is_palindrome(s[1:-1])

test = IsPalindrome()
print(test.is_palindrome("abcba"))
print(test.is_palindrome("abccba"))
print(test.is_palindrome("abcdba"))
print(test.is_palindrome("abccbb"))
print(test.is_palindrome("a"))
print(test.is_palindrome(""))